packages/ketcher-core/src/application/render/raphaelRender.ts

```diff
@@ -0,0 +1,279 @@
+    this.update(true)
+  }
+
+  setPaperSize(sz: Vec2) {
+    this.sz = sz
+    this.paper.setSize(sz.x * this.options.zoom, sz.y * this.options.zoom)
+    this.setViewBox(this.options.zoom)
+  }
+
+  setZoom(zoom: number) {
+    // when scaling the canvas down it may happen that the scaled canvas is smaller than the view window
+    // don't forget to call setScrollOffset after zooming (or use extendCanvas directly)
+    this.options.zoom = zoom
+    this.paper.setSize(this.sz.x * zoom, this.sz.y * zoom)
+    this.setViewBox(zoom)
+  }
+
+  setOffset(newoffset: Vec2) {
+    const delta = new Vec2(
+      newoffset.x - this.options.offset.x,
+      newoffset.y - this.options.offset.y
+    )
+    this.clientArea.scrollLeft += delta.x
+    this.clientArea.scrollTop += delta.y
+    this.options.offset = newoffset
+  }
+
+  setMolecule(struct: Struct) {
+    this.paper.clear()
+    this.ctab = new ReStruct(struct, this)
+    this.options.offset = new Vec2()
+    this.update(false)
+  }
+
+  setScrollOffset(x: number, y: number) {
+    const clientArea = this.clientArea
+    const cx = clientArea.clientWidth
+    const cy = clientArea.clientHeight
+    const e = calcExtend(
+      this.sz.scaled(this.options.zoom),
+      x,
+      y,
+      cx + x,
+      cy + y
+    ).scaled(1 / this.options.zoom)
+    if (e.x > 0 || e.y > 0) {
+      this.setPaperSize(this.sz.add(e))
+      const d = new Vec2(x < 0 ? -x : 0, y < 0 ? -y : 0).scaled(
+        1 / this.options.zoom
+      )
+      if (d.x > 0 || d.y > 0) {
+        this.ctab.translate(d)
+        this.setOffset(this.options.offset.add(d))
+      }
+    }
+
+    clientArea.scrollLeft = x * this.options.scale
+    clientArea.scrollTop = y * this.options.scale
+  }
+
+  update(force = false, viewSz: Vec2 | null = null) {
+    viewSz =
+      viewSz ||
+      new Vec2(
+        this.clientArea.clientWidth || 100,
+        this.clientArea.clientHeight || 100
+      )
+
+    const changes = this.ctab.update(force)
+    this.ctab.setSelection() // [MK] redraw the selection bits where necessary
+
+    if (changes) {
+      const boundingBox = this.ctab
+        ?.getVBoxObj()
+        ?.transform(Scale.obj2scaled, this.options)
+        .translate(this.options.offset || new Vec2())
+
+      if (!this.options.autoScale && boundingBox) {
+        const ext = Vec2.UNIT.scaled(this.options.scale)
+        const extendedBox =
+          boundingBox.sz().length() > 0
+            ? boundingBox.extend(ext, ext)
+            : boundingBox
+        const viewBox = new Box2Abs(
+          this.getScrollPos(),
+          viewSz.scaled(1 / this.options.zoom).sub(Vec2.UNIT.scaled(20))
+        )
+        const cb = Box2Abs.union(viewBox, extendedBox)
+        if (!this.oldCb) {
+          this.oldCb = new Box2Abs()
+        }
+
+        const sz = cb.sz().floor()
+        const delta = this.oldCb.p0.sub(cb.p0).ceil()
+        // TODO: do we need this.oldBb? Seem it's not in use
+        // this.oldBb = bb
```

Comment:
looks like commented-out code 😄

Reply:
Yes - a bit scary to remove anything completely - probably, it was needed for something